provider/internal/cdfmc/resource.go

```diff
@@ -0,0 +1,160 @@
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "TODO",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (r *Resource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*cdoClient.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *cdoClient.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *Resource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	tflog.Trace(ctx, "read cdfmc resource")
+
+	// 1. read state data
+	var stateData ResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &stateData)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// 2. do read
+	if err := Read(ctx, r, &stateData); err != nil {
+		resp.Diagnostics.AddError("failed to read Example resource", err.Error())
+	}
+
+	// 3. save data into terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateData)...)
+}
+
+func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, res *resource.CreateResponse) {
+
+	tflog.Trace(ctx, "create cdfmc resource")
+
+	// 1. read plan data into planData
+	var planData ResourceModel
+	res.Diagnostics.Append(req.Plan.Get(ctx, &planData)...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	// 2. use plan data to create device and fill up rest of the model
+	if err := Create(ctx, r, &planData); err != nil {
+		res.Diagnostics.AddError("failed to create Example resource", err.Error())
+		return
+	}
+
+	// 3. set state using filled model
+	res.Diagnostics.Append(res.State.Set(ctx, &planData)...)
+}
+
+func (r *Resource) Update(ctx context.Context, req resource.UpdateRequest, res *resource.UpdateResponse) {
+
+	tflog.Trace(ctx, "update cdfmc resource")
+
+	// 1. read plan data
+	var planData ResourceModel
+	res.Diagnostics.Append(req.Plan.Get(ctx, &planData)...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	// 2. read state data
+	var stateData ResourceModel
+	res.Diagnostics.Append(req.State.Get(ctx, &stateData)...)
+	if res.Diagnostics.HasError() {
+		return
+	}
+
+	// 3. do update
+	if err := Update(ctx, r, &planData, &stateData); err != nil {
+		res.Diagnostics.AddError("failed to update Example resource", err.Error())
+	}
+
+	// 4. set resulting state
+	res.Diagnostics.Append(res.State.Set(ctx, &stateData)...)
+}
+
+func (r *Resource) Delete(ctx context.Context, req resource.DeleteRequest, res *resource.DeleteResponse) {
+
+	tflog.Trace(ctx, "delete cdfmc resource")
```

Comment:
deletion is a noop, so do we really need to do an err check here? Do we even need an operation? It would be way clearer if you returned the tflog.warn here.

Reply:
Yea make sense